data/test/scenarios/swarm_disables_upgrades_with_abilities_adjacent.cfg

```diff
@@ -0,0 +1,199 @@
+    x,y=12,4
+    type=Elvish Hero
+    side=1
+    [/unit]
+    [unit]
+    id=ben
+    name=_"Ben"
+    x,y=14,3
+    type=Orcish Warrior
+    side=2
+[/unit]
+[modify_unit]
+    [filter]
+    [/filter]
+    max_hitpoints=100
+    hitpoints=100
+    attacks_left=1
+[/modify_unit]
+[object]
+    silent=yes
+    [effect]
+        apply_to=new_ability
+        [abilities]
+            [swarm]
+                swarm_attacks_max=1
+                swarm_attacks_min=1
+                {FILTER_BOB}
+            [/swarm]
+            [attacks]
+                value=10
+                {FILTER_BOB}
+            [/attacks]
+            [attacks]
+                add=13
+                {FILTER_BOB}
+            [/attacks]
+            [damage]
+                value=1
+                {FILTER_BOB}
+            [/damage]
+            [chance_to_hit]
+                value=100
+                {FILTER_BOB}
+            [/chance_to_hit]
+        [/abilities]
+    [/effect]
+    [filter]
+        id=ben
+    [/filter]
+[/object]
+[object]
+    silent=yes
+    [effect]
+        apply_to=new_ability
+        [abilities]
+            [attacks]
+                value=10
+                {FILTER_ALICE}
+            [/attacks]
+            [damage]
+                value=1
+                {FILTER_ALICE}
+            [/damage]
+            [chance_to_hit]
+                id={SPECIAL_ID}
+                value=100
+                {FILTER_ALICE}
+            [/chance_to_hit]
+        [/abilities]
+    [/effect]
+    [filter]
+        id=alex
+    [/filter]
+[/object]
+[object]
+    silent=yes
+    [filter]
+        id=bob
+    [/filter]
+    [effect]
+        apply_to=attack
+        increase_attacks=12
+    [/effect]
+[/object]
+
+[store_unit]
+    [filter]
+        id=alice
+    [/filter]
+    variable=a
+    kill=yes
+[/store_unit]
+[store_unit]
+    [filter]
+        id=bob
+    [/filter]
+    variable=b
+[/store_unit]
+[unstore_unit]
+    variable=a
+    find_vacant=yes
+    x,y=$b.x,$b.y
+[/unstore_unit]
+[store_unit]
+    [filter]
+        id=alice
+    [/filter]
+    variable=a
+[/store_unit]
+
+[do_command]
+    [attack]
+        weapon=0
+        defender_weapon=0
+        [source]
+            x,y=$a.x,$a.y
+        [/source]
+        [destination]
+            x,y=$b.x,$b.y
+        [/destination]
+    [/attack]
+[/do_command]
+[store_unit]
+    [filter]
+        id=alice
+    [/filter]
+    variable=a
+[/store_unit]
+[store_unit]
+    [filter]
+        id=bob
+    [/filter]
+    variable=b
+[/store_unit]
+{ASSERT ({VARIABLE_CONDITIONAL a.hitpoints equals 99})}
+{ASSERT ({VARIABLE_CONDITIONAL b.hitpoints equals 90})}
+{SUCCEED}
+#enddef
+
+{GENERIC_UNIT_TEST "swarm_disables_upgrades_with_abilities_adjacent" (
+    [event]
+        name=start
+        {ALICE_ATTACK_BOB test_cth}
+    [/event]
+)}
+
+{GENERIC_UNIT_TEST "swarm_disables_upgrades_with_abilities_adjacent_fail" (
+    [event]
+        name=start
+        {ALICE_ATTACK_BOB test_cth_fail}

```

Comment:
I think this is testing that an ability with a different name doesn't get matched by the filter for `test_cth`. Please move the final `{ASSERT ...}` and `{SUCCEED}` outside of the macro, and test exactly the values that are expected when the attack does get upgraded instead of testing a `fail`.

* That makes it a more rigorous test, checking the expected result exactly.
* It makes it clearer what has broken if it fails.
* The test runs a bit faster - all the "returns 0" tests in wml_test_schedule are run together, but the other ones need a separate instance of Wesnoth for each one.

Reply:
i had reverted add test fail